faststream/kafka/testing.py

```diff
@@ -91,17 +91,21 @@ async def publish(  # type: ignore[override]
             reply_to=reply_to,
         )
 
-        for handler in self.broker._subscribers.values():  # pragma: no branch
-            if topic in handler.topics:
-                return await call_handler(
-                    handler=handler,
-                    message=[incoming]
-                    if isinstance(handler, AsyncAPIBatchSubscriber)
-                    else incoming,
-                    rpc=rpc,
-                    rpc_timeout=rpc_timeout,
-                    raise_timeout=raise_timeout,
-                )
+        for consumer_group in set(
+            {sub.group_id for sub in self.broker._subscribers.values()}
```

Comment:
I don't like O(n**2) in this case. Can we find any decision without iteration throw all subscribers twice?